util/stmtsummary/evicted.go

```diff
@@ -0,0 +1,318 @@
+package stmtsummary
+
+import (
+	"container/list"
+	"time"
+
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/types"
+)
+
+// stmtSummaryByDigestEvicted contents digests evicted from stmtSummaryByDigestMap
+type stmtSummaryByDigestEvicted struct {
+	// record evicted data in intervals
+	// latest history data is Back()
+	history *list.List
+}
+
+// element being stored in stmtSummaryByDigestEvicted
+type stmtSummaryByDigestEvictedElement struct {
+	// *Kinds* of digest being evicted
+	digestKeyMap map[string]struct{}
+
+	// summary of digest being evicted
+	sum *stmtSummaryByDigestElement
+}
+
+// spawn a new pointer to stmtSummaryByDigestEvicted
+func newStmtSummaryByDigestEvicted() *stmtSummaryByDigestEvicted {
+	return &stmtSummaryByDigestEvicted{
+		history: list.New(),
+	}
+}
+
+// spawn a new pointer to stmtSummaryByDigestEvictedElement
+func newStmtSummaryByDigestEvictedElement(beginTimeForCurrentInterval int64, intervalSeconds int64) *stmtSummaryByDigestEvictedElement {
+	ssElement := new(stmtSummaryByDigestElement)
+	ssElement.beginTime = beginTimeForCurrentInterval
+	ssElement.endTime = beginTimeForCurrentInterval + intervalSeconds
+	return &stmtSummaryByDigestEvictedElement{
+		digestKeyMap: make(map[string]struct{}),
+		sum:          ssElement,
+	}
+}
+
+// AddEvicted is used add an evicted record to stmtSummaryByDigestEvicted
+func (ssbde *stmtSummaryByDigestEvicted) AddEvicted(evictedKey *stmtSummaryByDigestKey, evictedValue *stmtSummaryByDigest, historySize int) {
+
+	// *need to get optimized*!!
```

Comment:
Remove the blank line and refine the comment. Need to optimize for what?

Reply:
currently, I simply check through all elements in evicted digest and history list.

Assume the length of evicted digest is 'm', length of history list is 'n'. then time consumed on this operation is O(mn). 

This can be improved to O(m+n).